src/d/actor/d_a_obj_pbco.cpp

```diff
@@ -4,55 +4,144 @@
 //
 
 #include "d/actor/d_a_obj_pbco.h"
-#include "dolphin/types.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_item.h"
+#include "d/d_procname.h"
+
+static daObj_Pbco_HIO_c l_HIO;
+const char daObj_Pbco_c::M_arcname[] = "Pbco";
 
 /* 000000EC-00000104       .text __ct__16daObj_Pbco_HIO_cFv */
 daObj_Pbco_HIO_c::daObj_Pbco_HIO_c() {
-    /* Nonmatching */
+    field_0x04 = -1;
 }
 
 /* 00000104-0000018C       .text set_mtx__12daObj_Pbco_cFv */
 void daObj_Pbco_c::set_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(scale);
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::ZXYrotM(shape_angle);
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
 }
-
 /* 0000018C-000001AC       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static int CheckCreateHeap(fopAc_ac_c* i_this) {
+    daObj_Pbco_c* a_this = static_cast<daObj_Pbco_c*>(i_this);
+    return a_this->CreateHeap();
 }
 
 /* 000001AC-00000348       .text CreateHeap__12daObj_Pbco_cFv */
-void daObj_Pbco_c::CreateHeap() {
-    /* Nonmatching */
+BOOL daObj_Pbco_c::CreateHeap() {
+    int EndlessNightFlag;
+    J3DModelData* modelData;
+
+    EndlessNightFlag = dComIfGs_isEventBit(0x0a02);
```

Comment:
Does htis need to be in a separate variable? Try including it inline in the if statement.

Reply:
👍